### PR TITLE
d25/nye fixes

### PR DESCRIPTION
**Pull request by jmwall24**
Various fixes of d25/nye mishaps

## Fixes
- issue in `mas_d25_spent_time_monika` where the post-poem dialogue was specific to the first d25 poem and didn't really make sense with the other poems
- `persistent._mas_pm_accomplished_resolution` and `persistent._mas_pm_has_new_years_res` were being stored as `pm` vars instead of `nye` vars and thus not being reset on a yearly basis, leading to dialogue that didn't make sense. They are being converted to `persistent._mas_nye_accomplished_resolution` and `persistent._mas_nye_has_new_years_res` respecitvely.
- people who got the nye_dress previously and are now at `love` get `mas_nye_monika_nye_dress_intro` pushed but don't actually see anything related to the event since they already have the dress unlocked and we don't have Monika wear it without being asked in this case. This is now being changed so she puts it on in all cases if this event is pushed.
- `mas_nye_monika_nye_dress_intro` also had a bug for people who were getting the dress put on for a second+ time which has now been fixed (#8538)

## TODO
- [x] update script to delete the old `pm` vars and migrate data to new `nye` vars

### CODE REVIEW

**Review by ThePotatoGuy — DISMISSED** [review]

**Review by ThePotatoGuy — APPROVED** [review]

**Comment by Booplicate** [comment]
We might need to transfer old history from `pm.actions.made_new_years_resolutions` to `nye.actions.made_new_years_resolutions`. Right you have to check both to get the correct result.

**Comment by ThePotatoGuy** [comment]
added transferring historial data in the update script

**Review by ThePotatoGuy — APPROVED** [review]

File: Monika After Story/game/updates.rpy [hunk 1]
```diff
@@ -381,9 +381,18 @@ label v0_12_8_1(version="v0_12_8_1"):
...
+                "nye.actions.did_new_years_resolutions"
+            )
             safeDel("_mas_pm_accomplished_resolutions")
+
         if persistent._mas_nye_has_new_years_res is None:
             persistent._mas_nye_has_new_years_res = persistent._mas_pm_has_new_years_res
+            store.mas_history._store_all(
+                mas_HistLookup_all("pm.actions.made_new_years_resolutions"),
+                "nye.actions.made_new_years_resolutions"
+            )
```
Comment (Booplicate) [inline Monika After Story/game/updates.rpy:395]: `mas_HistLookup_all` returns data of format `{year: (code, value)}`,
but `_store_all` uses `_store` which accepts `value`, `key`, `year`. This code will save a tuple `(code, value)` as `value`.

Comment (Booplicate) [reply Monika After Story/game/updates.rpy:395]: Something like `{k: v[1] for k, v in mas_HistLookup_all(KEY).items()}` would work I think, unless you know a better way.

Comment (ThePotatoGuy) [reply Monika After Story/game/updates.rpy]: `_store_all` says its supposed to work with the data tuple from `mas_HistLookup` so I changed `_store_all` instead.

**Review by Booplicate — APPROVED** [review]

**Review by multimokia — APPROVED** [review]

**Review by Legendkiller21 — APPROVED** [review]

**Pull request merged**